packages/server/src/api/routes/tests/search.spec.ts

```diff
@@ -76,13 +91,190 @@ describe.each([
     it.each(stringSearchTests)(
       `should be able to run query: $query`,
       async ({ query, expected }) => {
-        await Promise.all(rows.map(r => config.api.row.save(table._id!, r)))
+        const savedRows = await Promise.all(
+          rows.map(r => config.api.row.save(table._id!, r))
+        )
+        const { rows: foundRows } = await config.api.row.search(table._id!, {
+          tableId: table._id!,
+          query,
+        })
+        expect(foundRows).toEqual(
+          expect.arrayContaining(
+            expected.map(r =>
+              expect.objectContaining(savedRows.find(sr => sr.name === r.name)!)
+            )
+          )
+        )
+      }
+    )
```

Comment:
Changed the body of each of these functions to assert that the result contains the body of the _saved_ row, not the bare `{ name: "whatever" }`. Slightly more robust, and in the case of dates lower down they're saved as `Date` objects but return as ISO `string`s. So this gives some resilience against conversions, which I'm sure will come up with some other data types.